src/enterprise-edge/azext_enterprise_edge/custom.py

```diff
@@ -52,18 +52,25 @@ def get(cls, cmd, resource_group, name):
         return cls._request(cmd, resource_group, name)
 
     @classmethod
-    def _validate_cdn_provider_registered(cls, cmd):
+    def _register_cdn_provider(cls, cmd):
         management_hostname = cmd.cli_ctx.cloud.endpoints.resource_manager
         api_version = "2021-04-01"
         sub_id = get_subscription_id(cmd.cli_ctx)
-        url_fmt = "{}/subscriptions/{}/providers/Microsoft.CDN?api-version={}"
+        url_fmt = "{}/subscriptions/{}/providers/Microsoft.CDN/register?api-version={}"
 
         request_url = url_fmt.format(management_hostname.strip('/'), sub_id, api_version)
 
-        registration = send_raw_request(cmd.cli_ctx, "GET", request_url).json().get("registrationState").lower()
-        if registration != "registered":
-            raise CLIError("Provider Microsoft.CDN is not registered. "
-                           "Please run 'az provider register --wait --namespace Microsoft.CDN'")
+        try:
```

Comment:
This is ok for now since this is in an extension - but this should be changed to using the SDK call directly & handling case for error without the try catch block.